### ISSUE TITLE
Interval equality is incorrect for perfect unisons

### ISSUE THREAD

**Issue by jeremymui** [cuthbertLab/music21#340]
```
>>> c = note.Note('c4')
>>> d = note.Note('d4')
>>> interval.Interval(c, d)
<music21.interval.Interval M2>
>>> interval.Interval(c, d) == interval.Interval('M2')
True
>>> interval.Interval(d, c)
<music21.interval.Interval M-2>
>>> interval.Interval(d, c) == interval.Interval('M-2')
True
>>> interval.Interval(c, c)
<music21.interval.Interval P1>
>>> interval.Interval(c, c) == interval.Interval('P1')
False
>>> interval.Interval(c, c) == interval.Interval('P-1')
False
```

**Comment by jeremymui** [comment]
The `__eq__` method in line 2089 checks if the diatonic forms of the two intervals are the same, and they are not.
https://github.com/cuthbertLab/music21/blob/b58aa3d455aea2568ae5303c47687a4b06f9ec26/music21/interval.py#L2089-L2091
```
>>> c = note.Note('c4')
>>> interval1 = interval.Interval(c, c)
>>> interval2 = interval.Interval('P1')
>>> interval1.diatonic
<music21.interval.DiatonicInterval P1>
>>> interval2.diatonic
<music21.interval.DiatonicInterval P1>
>>> interval1.diatonic == interval2.diatonic
False
```
This is because their directions are different, but they should be the same.
```
>>> interval1.diatonic.direction
<Direction.OBLIQUE: 0>
>>> interval2.diatonic.direction
<Direction.ASCENDING: 1>
```

**Comment by mscuthbert** [comment]
Fixed in 31319ca
